README.md

```diff
@@ -73,14 +73,25 @@ This diagram covers the major classes of transformations. The most basic ones ar
 Here is a very simple example of an algebra (`eval`) and how to apply it to a recursive structure.
 
 ```scala
+
+// we will need a Functor[Expr] in order to call embedT bellow
+implicit val exprFunctor = new scalaz.Functor[Expr] {
+  override def map[A, B](fa: Expr[A])(f: (A) => B) = fa match{
+    case Num(value) => Num[B](value)
+    case Mul(l, r) => Mul(f(l), f(r))
+  }
+}
+
 val eval: Algebra[Expr, Long] = { // i.e. Expr[Long] => Long
   case Num(x)    => x
   case Mul(x, y) => x * y
 }
+ 
+def someExpr[T[_[_]]: CorecursiveT]: T[Expr] =
+  Mul(Num[T[Expr]](2).embedT, Mul(Num[T[Expr]](3).embedT,
+      Num[T[Expr]](4).embedT).embedT).embedT
```

Comment:
I've seen the comments above `CorecursiveT` so I suspected it was a sub-optimal solution.
I'll do my best making sense out of all the words you've written there (it's still a bot hard for me), but meanwhile I'll test teh solution you propose on my local project (manual tut eh) and modify the PR.